app/uk/gov/hmrc/transitmovementsrouter/services/StreamingMessageTrimmer.scala

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright 2022 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.transitmovementsrouter.services
+
+import akka.stream.alpakka.xml.scaladsl.XmlWriting
+import akka.stream.alpakka.xml.scaladsl.XmlParsing
+import akka.stream.scaladsl.Source
+import akka.util.ByteString
+import com.google.inject.ImplementedBy
+import com.google.inject.Inject
+import uk.gov.hmrc.transitmovementsrouter.models.MessageType
+
+@ImplementedBy(classOf[StreamingMessageTrimmerImpl])
+trait StreamingMessageTrimmer {
+
+  def trim(source: Source[ByteString, _], messageType: MessageType): Source[ByteString, _]
+}
+
+class StreamingMessageTrimmerImpl @Inject() extends StreamingMessageTrimmer {
+
+  override def trim(source: Source[ByteString, _], messageType: MessageType): Source[ByteString, _] =
+    source
+      .via(XmlParsing.parser)
+      .via(XmlParsing.subslice("TraderChannelResponse" :: messageType.rootNode :: Nil))
```

Comment:
So this has changed, since this comment (it sliced wrong, whoops) - but essentially the TraderChannelResponse tag is entirely andEIS wrapper, NCTS doesn't include it, we don't care to keep it and we don't intend to forward it along to the third parties in any way, so we remove it before we forward it to be persisted or pushed to the third parties.